### DISCUSSION TITLE
Hacktoberfest 2024! 🎃 👻

### DISCUSSION THREAD

**Discussion by mikaelacaron** [mikaelacaron/Basic-Car-Maintenance#303] [Announcements]
# Hacktoberfest 2024 is here!

New this year! Using the [Firebase Local Emulator](https://firebase.google.com/docs/emulator-suite) for development, as opposed to the production Firebase, which I'll be deleting all the data that is currently saved there, before I ship to the App Store (because to this point, it's all only test data)

New steps in CONTRIBUTING.md for setting up the Firebase Local Emulator, and what you need to download to make it work.
Feel free to start a new Discussions thread if you have any issues, or comment below here, and if you want, create a PR to update the instructions to help everyone!

New issues will be created throughout the month, if you don't see any, or want to request a new feature, go ahead and open an issue.

This year it's recommended to use Xcode 16.0! We'll start implementing new features from iOS 18!

The goals of this month is of course to continue the app, add iOS 18 features, and ship it! We'll think of new features as we go!

Most conversations should happen in [Discussions](https://github.com/mikaelacaron/Basic-Car-Maintenance/discussions), and I would encourage you to start there, but if you would like, you can still DM me on social media ([Mikaela's social links](https://mikaelacaron.com/links)) to suggest a new feature or if you have issues.

# Happy Hacktoberfest!

Thank you for contributing! ❤️ 

- Mikaela Caron

**Comment by maartinj** [comment]
I am not 100% whether does it have anything to do with using firebase local emulator, but as far as I noticed:
- On 1st launch, loading circle is still running in the main screen. After 2nd-3rd app launch, it started to working fine (no loading circle, just the info 'Tap the + to begin')
- When added 1st vehicle, was still not available in 'AddMaintenanceView'. After 2nd-3rd app launch, it started to be available
- 'Change app icon' do not show the icons (attached screenshot)
<img src="https://github.com/user-attachments/assets/e41d34d1-2317-40e0-a5e2-c47ff2757217" height="500">

**Reply by mikaelacaron** [reply]
# File these as multiple separate issues, if you can reproduce it

> On 1st launch, loading circle is still running in the main screen. After 2nd-3rd app launch, it started to working fine (no loading circle, just the info 'Tap the + to begin')

#288 should've fixed this, but maybe it didn't?

> When added 1st vehicle, was still not available in 'AddMaintenanceView'. After 2nd-3rd app launch, it started to be available

Add the first vehicle and then do you see it in the Firebase Firestore emulator, (before going to `AddMaintenanceView`, then navigate to `AddMaintenanceView`, is it still not showing?
Sounds like a bug

> 'Change app icon' do not show the icons (attached screenshot)

I haven't seen this one...

**Reply by maartinj** [reply]
Looks like, when Onboarding process will be done and implemented, loading circle is not available and 1st vehicle will be added and properly visible. Magic 🪄

For 'Change app icon', I can create issue 👍
